sandbox/git-annex-remote-gitobjects

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+#	bash - for string indexing
+#
+# git-annex external special remote prototype to test feasibility of
+# using regular git objects and refs to store content of files in remote
+# git objects store.
+#
+#
+# Copyright 2013 Joey Hess; licenced under the GNU GPL version 3 or higher.
+#  tuned for the gitobjects type remote by Yaroslav Halchenko, 2019
+
+set -e
+
+# This program speaks a line-based protocol on stdin and stdout.
+# When running any commands, their stdout should be redirected to stderr
+# (or /dev/null) to avoid messing up the protocol.
+runcmd () {
+	echo DEBUG "Running $@"
+	"$@" >/dev/null 2>&1 # &2
+}
+
+# Prevent any gc so we could avoid objects being packed before we get
+# rid of them
+rungitcmd () {
+	runcmd git -c gc.auto=0 "$@"
+}
+
+# Gets a value from the remote's configuration, and stores it in RET
+getconfig () {
+	ask GETCONFIG "$1"
+}
+
+# Stores a value in the remote's configuration.
+setconfig () {
+	echo SETCONFIG "$1" "$2"
+}
+
+# Sets ref to the reference to use to store a key.
+calcref () {
+	ref="refs/annexed/$1"
+}
+
+calcobjfile () {
+	objfile="$dotgit/objects/${h:0:2}/${h:2}"
+	echo DEBUG "objfile=$objfile"
+}
+
+# Asks for some value, and stores it in RET
+ask () {
+	echo "$1" "$2"
+	read resp
+	# Tricky POSIX shell code to split first word of the resp,
+	# preserving all other whitespace
+	case "${resp%% *}" in
+		VALUE)
+			RET="$(echo "$resp" | sed 's/^VALUE \?//')"
+		;;
+		*)
+			RET=""
+		;;
+	esac
+}
+
+cleanup () {
+	# remove reference and object locally
+	rm -f "$dotgit/$ref"
```

Comment:
`git update-ref -d "$ref"` or `git update-ref -d "$ref" "$h"`?

Reply:
But perhaps you can get away with not creating a local ref at all:  `git push <url> "$h":"$ref"` and `git fetch <url> "$h"`.

Reply:
oh... I could do that? ;)  I will give it a shot later on! thanks